functions-and-scope.js

```diff
@@ -42,25 +63,50 @@ const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
 
 // ---- Verwachte uitkomst: 6.642857142857143
 
+let summedGrade = 0;
+for (let i = 0; i < grades.length; i++) {
+    summedGrade += grades[i];
+}
+
+console.log(summedGrade / grades.length);
 
 /* 2b: Omschrijven tot een herbruikbare functie */
 // Schrijf een functie genaamd averageGrade, die een array van cijfers verwacht (zoals grades) en het gemiddelde cijfer teruggeeft. Gebruik hiervoor jouw antwoord van 2a.
 // Zorg ervoor dat jouw functie ook werkt als we een andere array willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
 // Log het antwoord in de terminal.
 
+
+function averageGrade(gradesArray) {
+    let summedGrade = 0;
+    for (let i = 0; i < gradesArray.length; i++) {
+        summedGrade += gradesArray[i];
+    }
+    return summedGrade / gradesArray.length;
+}
+
+console.log(averageGrade(grades));
+
 // ---- Verwachte uitkomsten:
 // averageGrade(grades) geeft 6.642857142857143
 // averageGrade([6, 4, 5]) geeft xxxx
-// averageGrade([8, 9, 4, 6, 10]) geeft xxxx
+// averageGrade([8, 9, 4, 6, 10]) geeft xxxxL
 
+function averageGrade2Dec(gradesArray) {
+    let summedGrade = 0;
+    for (let i = 0; i < gradesArray.length; i++) {
+        summedGrade += gradesArray[i];
+    }
+    // https://stackoverflow.com/questions/11832914/how-to-round-to-at-most-2-decimal-places-if-necessary
+    return +(Math.round(summedGrade / gradesArray.length + "e+2") + "e-2");
```

Comment:
Dit kan wat simpeler:
```suggestion
    return (summedGrade / gradesArray.length).toFixed(2);
```

Reply:
Dat is zeker simpeler, alleen geeft deze een foute afronding bij bijv. : 1.005. Dat zou natuurlijk 1.01 moeten zijn maar bij de oplossing die jij aangeeft is dat 1.00 Dat is meestal niet een probleem, maar als je moeilijkere berekeningen maakt da kan dat onverwachte fouten opleveren.